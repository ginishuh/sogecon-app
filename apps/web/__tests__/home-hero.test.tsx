import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';

import HomePage from '../app/page';

describe('HomePage hero and cards', () => {
  it('renders hero carousel with dots and caption', () => {
    render(<HomePage />);
    expect(screen.getByRole('heading', { name: '서강대 경제대학원 총동문회' })).toBeInTheDocument();
    // 캡션 바 존재
    expect(screen.getByText(/공지 · 행사 · 동문 수첩을 한 곳에서|총동문회 웹 런치/)).toBeInTheDocument();
    // 도트 영역 존재(폴백 1~2개 중 하나 이상)
    const dots = document.querySelectorAll('.hero-dot');
    expect(dots.length).toBeGreaterThan(0);
  });

  it('renders quick access cards with metadata and about promo card', () => {
    render(<HomePage />);
    const cards = screen.getAllByRole('link', { name: /·/ });
    expect(cards).toHaveLength(3);
    expect(screen.getByRole('link', { name: '동문 수첩 · 동문 수첩 베타 공개' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: '가이드 · 웹 런치 FAQ 12선' })).toBeInTheDocument();
    // About promo card exists
    expect(screen.getByRole('link', { name: '총동문회 소개 · 인사말·연혁·조직' })).toBeInTheDocument();
  });
});
