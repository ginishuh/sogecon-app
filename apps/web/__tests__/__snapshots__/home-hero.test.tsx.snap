// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage hero and cards > renders hero title, description, and primary CTA 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-16"
  >
    <section
      aria-labelledby="home-hero"
      class="home-hero"
    >
      <div>
        <span
          class="home-hero__eyebrow"
        >
          2025 웹 런치
        </span>
        <h1
          class="home-hero__title"
          id="home-hero"
        >
          한 번의 로그인으로 동문 네트워크 전체를
        </h1>
        <p
          class="home-hero__description"
        >
          공지, 행사, 디렉터리, 소개 페이지를 하나의 홈에서 연결했습니다. 모바일에서도 동일한 경험을 제공하며, 2025년 4분기부터는 푸시 알림과 회원 전용 기능을 순차적으로 공개합니다.
        </p>
        <div
          aria-label="주요 행동"
          class="home-hero__actions"
          role="group"
        >
          <a
            class="home-hero__cta"
            href="/directory"
          >
            동문 디렉터리 열기
          </a>
          <a
            class="home-hero__secondary"
            href="/faq"
          >
            자주 묻는 질문
          </a>
        </div>
        <dl
          aria-label="2025년 하이라이트 지표"
          class="mt-6 grid gap-4 text-sm text-neutral-muted md:grid-cols-2"
        >
          <div>
            <dt
              class="text-xs uppercase tracking-widest text-neutral-muted"
            >
              다음 행사
            </dt>
            <dd
              class="font-semibold text-neutral-ink"
            >
              정기총회 & 런치 데이 · 10월 25일(토) 14:00
            </dd>
          </div>
          <div>
            <dt
              class="text-xs uppercase tracking-widest text-neutral-muted"
            >
              공지 업데이트
            </dt>
            <dd
              class="font-semibold text-neutral-ink"
            >
              FAQ/정책 문서 최종본 · 10월 8일 게시
            </dd>
          </div>
        </dl>
      </div>
      <div
        class="home-hero__image"
      >
        <img
          alt="웹 런치를 기념하는 총원우회 일러스트"
          class="h-auto w-full rounded-2xl object-cover shadow-soft"
          height="360"
          sizes="(max-width: 768px) 100vw, 480px"
          src="/images/home/hero-launch.svg"
          width="480"
        />
      </div>
    </section>
    <section
      aria-labelledby="home-stats"
      class="rounded-3xl bg-brand-surface px-6 py-8 shadow-sm md:px-10"
    >
      <div
        class="flex flex-col gap-6 md:flex-row md:items-center md:justify-between"
      >
        <div>
          <span
            class="text-xs font-semibold uppercase tracking-widest text-neutral-muted"
          >
            총원우회 스냅숏
          </span>
          <h2
            class="mt-2 font-heading text-2xl text-neutral-ink md:text-3xl"
            id="home-stats"
          >
            2025년 4분기 준비 현황
          </h2>
        </div>
        <a
          class="home-hero__secondary"
          href="/about/org"
        >
          조직도 자세히 보기
        </a>
      </div>
      <dl
        class="mt-8 grid gap-6 text-sm text-neutral-muted md:grid-cols-4"
      >
        <div
          class="rounded-2xl border border-neutral-border bg-white p-4 shadow-sm"
        >
          <dt
            class="text-xs font-semibold uppercase tracking-widest text-neutral-muted"
          >
            등록 회원
          </dt>
          <dd
            class="mt-2 text-xl font-semibold text-neutral-ink"
          >
            1,286명
          </dd>
        </div>
        <div
          class="rounded-2xl border border-neutral-border bg-white p-4 shadow-sm"
        >
          <dt
            class="text-xs font-semibold uppercase tracking-widest text-neutral-muted"
          >
            2025 주요 행사
          </dt>
          <dd
            class="mt-2 text-xl font-semibold text-neutral-ink"
          >
            17건 진행
          </dd>
        </div>
        <div
          class="rounded-2xl border border-neutral-border bg-white p-4 shadow-sm"
        >
          <dt
            class="text-xs font-semibold uppercase tracking-widest text-neutral-muted"
          >
            장학 기금
          </dt>
          <dd
            class="mt-2 text-xl font-semibold text-neutral-ink"
          >
            ₩196,500,000
          </dd>
        </div>
        <div
          class="rounded-2xl border border-neutral-border bg-white p-4 shadow-sm"
        >
          <dt
            class="text-xs font-semibold uppercase tracking-widest text-neutral-muted"
          >
            운영 프로젝트 팀
          </dt>
          <dd
            class="mt-2 text-xl font-semibold text-neutral-ink"
          >
            8개 준비
          </dd>
        </div>
      </dl>
    </section>
    <section
      aria-labelledby="home-updates"
      class="flex flex-col gap-6"
    >
      <div
        class="flex flex-col gap-3 md:flex-row md:items-center md:justify-between"
      >
        <div>
          <span
            class="text-xs font-semibold uppercase tracking-widest text-neutral-muted"
          >
            공지 & 행사
          </span>
          <h2
            class="font-heading text-2xl text-neutral-ink md:text-3xl"
            id="home-updates"
          >
            지금 확인할 소식
          </h2>
        </div>
        <a
          class="home-hero__secondary"
          href="/posts"
        >
          공지 전체 보기
        </a>
      </div>
      <div
        class="home-card-grid"
      >
        <a
          aria-label="디렉터리 · 동문 디렉터리 베타 공개"
          class="home-card"
          href="/directory"
        >
          <div>
            <span
              class="text-xs font-semibold uppercase tracking-widest text-brand-primary"
            >
              디렉터리
            </span>
            <h3
              class="home-card__title"
            >
              동문 디렉터리 베타 공개
            </h3>
            <p
              class="home-card__description"
            >
              검색, 기수/업종 필터, 정렬 기능까지 포함된 디렉터리를 웹에서 바로 이용하세요.
            </p>
          </div>
          <div
            class="home-card__meta"
          >
            <span>
              실시간 업데이트 중
            </span>
            <span>
              자세히
            </span>
          </div>
        </a>
        <a
          aria-label="행사 · 2025 정기총회 & 런치 데이"
          class="home-card"
          href="/events"
        >
          <div>
            <span
              class="text-xs font-semibold uppercase tracking-widest text-brand-primary"
            >
              행사
            </span>
            <h3
              class="home-card__title"
            >
              2025 정기총회 & 런치 데이
            </h3>
            <p
              class="home-card__description"
            >
              10월 25일 토요일, 다산관에서 열리는 오프라인+온라인 통합 런치 이벤트에 참여 신청하세요.
            </p>
          </div>
          <div
            class="home-card__meta"
          >
            <span>
              D-17 · 현장 + 온라인
            </span>
            <span>
              자세히
            </span>
          </div>
        </a>
        <a
          aria-label="가이드 · 웹 런치 FAQ 12선"
          class="home-card"
          href="/faq"
        >
          <div>
            <span
              class="text-xs font-semibold uppercase tracking-widest text-brand-primary"
            >
              가이드
            </span>
            <h3
              class="home-card__title"
            >
              웹 런치 FAQ 12선
            </h3>
            <p
              class="home-card__description"
            >
              로그인, 접근 권한, 개인정보 처리 방침 개편 내용을 Q&A 형식으로 정리했습니다.
            </p>
          </div>
          <div
            class="home-card__meta"
          >
            <span>
              업데이트: 2025-10-08
            </span>
            <span>
              자세히
            </span>
          </div>
        </a>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
